include/fe/fe_interface.h

```diff
@@ -109,6 +110,18 @@ class FEInterface
                                      const ElemType t,
                                      const unsigned int n);
 
+  typedef unsigned int (*n_dofs_at_node_ptr) (const ElemType,
```

Comment:
TODO for later maybe: there's a C++11 `using` syntax that improves the typedef'ing function pointers IMO. In your case it would read:
```
using n_dofs_at_node_ptr = unsigned int (*) (const ElemType, const Order, const unsigned int);
```
I don't think we have a configure test for it yet but it should be simple to add one.

Reply:
I've been using that syntax all over the place in MOOSE since we first switched to C++11. It's well supported. You probably don't need a configure test for every little piece of syntax. Yes there are still problems with things like `<regex>` but the syntax seems ok.

Reply:
Looks no better to me (although I do like the similar ```using foo = decltype(example_function);``` alternative), but we can use using if you prefer.

http://en.cppreference.com/w/cpp/compiler_support and https://gcc.gnu.org/projects/cxx-status.html claim that template aliases (which ought to be a dependant on other using declarations) went into gcc 4.7, which actually is later than any of the other features we specifically test for.  I'm not sure whether they got non-templated using declarations in first... but it looks like the whole thing was defined in the same [N2258](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2007/n2258.pdf) proposal, so I wouldn't bet on it.

On the gripping hand, I'm not sure if we *want* to do extra m4 work to help gcc 4.6 users decide it's finally time to upgrade; maybe they can figure that out on their own?